include/orc/dwarf_structs.hpp

```diff
@@ -68,7 +68,7 @@ struct attribute_value {
 
     void string(pool_string x) {
         _type |= type::string;
-        _string = std::move(x);
+        _string = x;
```

Comment:
Did the `move` here break things? Or is it just not necessary anymore?

Reply:
Just not necessary; the pool_string is just a pointer. (From a style / communication perspective, I'm not sure which is better.)

Reply:
It's just a pointer _for now_ 😃

What about using pass-by-rvalue-ref here, and then peppering the code with `copy` / `move` as appropriate? It'd probably take this PR in a direction you don't want it to go. In which case, can we track it as an issue?

Reply:
I assert it needs to be a pointer forever, else it isn't an interned string, and breaks the optimization. I did add:
```
static_assert(sizeof(pool_string) <= sizeof(intptr_t), "pool_string is design to be as small and fast to copy as a pointer.");
```
There's an argument that `pool_string` should be a typedef of a pointer (which makes the constraint clearer) but it is very useful to decorate it with methods.